lib/forwarder.js

```diff
@@ -1,15 +1,52 @@
 import net from 'node:net';
 import supportsColor from 'supports-color';
-import { removeConfig } from '../lib/config.js';
+import { LINT_COMMAND, PING_COMMAND, SHUTDOWN_COMMAND } from './commands.js';
+import { removeConfig } from './config.js';
 
 /**
- * @import { Config} from '../lib/config.js'
- * @import { Resolver} from '../lib/resolver.js'
+ * @import { Config } from '../lib/config.js'
+ * @import { Resolver } from '../lib/resolver.js'
  */
 
 const EXIT_TOKEN_REGEXP = new RegExp(/EXIT([0-9]{3})/);
 const EXIT_TOKEN_LENGTH = 7;
 
+/**
+ * @param {Config | null} config
+ * @returns {Promise<boolean>}
+ */
+export function isAlive(config) {
+  return forwardCommandToDaemon(config, PING_COMMAND)
+    .then(() => true)
+    .catch(() => false);
+}
+
+/**
+ * @param {Config | null} config
+ * @param {string} command
+ * @returns {Promise<void>}
+ */
+export async function forwardCommandToDaemon(config, command) {
+  if (!config) {
+    return Promise.reject(new Error('config not found'));
+  }
+
+  const socket = await connectToDaemon(config);
+  socket.write(JSON.stringify([config.token, command]));
+  if (command === SHUTDOWN_COMMAND) {
+    socket.destroySoon();
```

Comment:
I have no idea why `destroySoon` is needed, but without this method call daemon will never stop

Reply:
Hm. Maybe because we're calling `shutdown()` immediately which closes the server, and then we're calling `.end()` on the socket? Just a wild guess ...

Reply:
Yeah, as far as I understand this is related to server.close and half-open connections, for some reasons this connections could not be closed if server is closing, so I've implemented logic to await for connections completion, seems to work fine

Reply:
Also I've fixed all tests that was trivial to fix, 3 tests still failing, if you're ok with solution I'll continue fixing this tests

Reply:
After all tests are fixed I will continue with restart daemon logic in forwardToDaemon function